### PR TITLE
4459 Export report to Excel button

### LINKED ISSUE #4459: Reports: Export to Excel button

### PR DESCRIPTION
**Pull request by CarlosNZ**
Fixes #4459

# 👩🏻‍💻 What does this PR do?

Adds "Export" button on report page, with functionality to call the generateReport endpoint and download the generated file.

Have also added a common "downloadFile" method to FileUtils, which we can re-use from now on.

## 💌 Any notes for the reviewer?

For some reason this PR is showing changes that are already in the base branch. Please ignore the Rust files in here, they shouldn't be showing up.

See inline comments for additional details.

# 🧪 Testing

Need the "reports" repo checked out with the `add-excel-option` branch, and upsert the stock-status report definitions. https://github.com/msupply-foundation/open-msupply-reports/pull/74

Screenshot 2024-08-01 at 2 38 27 PM
- Go to `/reports` page. Click the "Stock And Items test" report.
- On the resulting report page, click the "Export" button up the top
- Excel file should be downloaded :)

### CODE REVIEW

File: client/packages/common/src/utils/files/FileUtils.ts [hunk 1]
```diff
@@ -25,7 +25,20 @@ const exportFile = (data: string, type: string, title?: string) => {
...
+  const data = await res.blob();
+  const header = res.headers.get('Content-Disposition');
+  const filename = header?.match(/filename="(.+)"/)?.[1];
+  const a = document.createElement('a');
+  a.href = window.URL.createObjectURL(data);
+  if (filename) a.download = filename;
+  a.click();
+  a.remove();
+};
+
```
Comment (CarlosNZ) [inline client/packages/common/src/utils/files/FileUtils.ts:39]: Common method for re-use

File: client/packages/system/src/Report/api/hooks/useGenerateReport.ts [hunk 2]
```diff
@@ -7,13 +7,13 @@ export const useGenerateReport = () => {
   const { reportApi, storeId } = useReportGraphQL();
 
   const mutationFn = async (params: GenerateReportParams) => {
-    const { dataId, reportId, args, sort } = params;
+    const { dataId, reportId, args, sort, format = PrintFormat.Html } = params;
```
Comment (CarlosNZ) [inline client/packages/system/src/Report/api/hooks/useGenerateReport.ts:10]: Re-used the same hook, but made `format` an (optional) parameter

File: client/packages/system/src/Report/api/hooks/usePrintReport.ts [hunk 3]
```diff
@@ -89,8 +91,7 @@ export const usePrintReport = () => {
       if (EnvUtils.printFormat === PrintFormat.Html) {
         printPage(url);
       } else {
-        const win = window.open(url, '_blank');
-        win?.focus();
+        FileUtils.downloadFile(url);
```
Comment (CarlosNZ) [inline client/packages/system/src/Report/api/hooks/usePrintReport.ts:94]: Replaced this with new "download" method for consistency.

File: server/Cargo.lock
Comment (CarlosNZ) [inline server/Cargo.lock:1]: Ignore all files from here on, they shouldn't appear.

**Review by roxy-dao — APPROVED** [review]
LGTM~ Thanks Carl

**Pull request merged**